src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/AssemblyReferenceHandler.cs

```diff
@@ -16,25 +16,45 @@ internal class AssemblyReferenceHandler : AbstractReferenceHandler
         { }
 
         protected override Task RemoveReferenceAsync(ConfiguredProjectServices services,
-            ProjectSystemReferenceInfo referencesInfo)
+            string itemSpecification)
         {
             Assumes.Present(services.AssemblyReferences);
 
             AssemblyName? assemblyName = null;
             string? assemblyPath = null;
 
-            if (Path.IsPathRooted((referencesInfo.ItemSpecification)))
+            if (Path.IsPathRooted(itemSpecification))
             {
-                assemblyPath = referencesInfo.ItemSpecification;
+                assemblyPath = itemSpecification;
             }
             else
             {
-                assemblyName = new AssemblyName(referencesInfo.ItemSpecification);
+                assemblyName = new AssemblyName(itemSpecification);
             }
 
             return services.AssemblyReferences.RemoveAsync(assemblyName, assemblyPath);
         }
 
+        protected override Task AddReferenceAsync(ConfiguredProjectServices services, string itemSpecification)
+        {
+            Assumes.Present(services.AssemblyReferences);
+
+            AssemblyName? assemblyName = null;
+            string? assemblyPath = null;
+
+            if (Path.IsPathRooted(itemSpecification))
+            {
+                assemblyPath = itemSpecification;
+            }
+            else
+            {
+                assemblyName = new AssemblyName(itemSpecification);
+            }
```

Comment:
I put this on the other PR as a suggestion:
```suggestion
            bool isPathRooted = Path.IsPathRooted(itemSpecification);
            AssemblyName? assemblyName = isPathRooted ? null : new AssemblyName(itemSpecification);
            string? assemblyPath = isPathRooted ? itemSpecification : null;
```

Reply:
I find the conditional dependence harder to see between the two ternaries here, compared to the original code.

If you really wanted to reduce this, perhaps:

```c#
(AssemblyName? assemblyName, string? assemblyPath) = Path.IsPathRooted(itemSpecification)
    ? (null, new AssemblyName(itemSpecification))
    : (itemSpecification, null);
```

But I'm not sure it's really better than the original.

Reply:
Again, it is preference. However, the ValueTuple version there (from my understanding) would do more allocations than just directly assigning the values. I tend to not create extraneous statement groups (if {} and else {}) unless absolutely necessary. I like binding variable declaration and initialization as close as possible when possible. Makes understanding (for me) more straightforward. And the extra statement groups encourages people (over time) to just inject random statements into them that have no relationship to their original intent. It is just a 'convenient' place to put it, and thus makes it more complex. I'm just very anti-imperative programming as I've seen it cause more bugs than it prevents. 😝

Reply:
> would do more allocations than just directly assigning the values

There are no heap allocations in the code I proposed, if that's what you're referring to.